src/som/vmobjects/SDouble.java

```diff
@@ -24,16 +24,21 @@
 
 package som.vmobjects;
 
+import com.oracle.truffle.api.CompilerDirectives;
 import com.oracle.truffle.api.interop.InteropLibrary;
 import com.oracle.truffle.api.library.ExportLibrary;
 import com.oracle.truffle.api.library.ExportMessage;
 
+import com.oracle.truffle.api.profiles.ValueProfile;
 import som.vm.Universe;
 
 
 @ExportLibrary(InteropLibrary.class)
 public final class SDouble extends SNumber {
 
+  @CompilerDirectives.CompilationFinal private final ValueProfile valueProfile =
+      ValueProfile.createClassProfile();
```

Comment:
Would drop this. This means that every single double number gets a separate profile. This seems very costly.